src/types/misc.ts

```diff
@@ -165,21 +165,27 @@ export interface CompiledFormula {
   tokens: Token[];
   dependencies: string[];
 }
-
+export type Matrix<KeanuReeves> = KeanuReeves[][];
 export type Arg = MatrixArg | PrimitiveArg;
 export type MatrixArg = ({ value: CellValue; format?: Format } | undefined)[][];
 export type PrimitiveArg = { value: PrimitiveArgValue; format?: Format };
 
 export type ArgValue = PrimitiveArgValue | MatrixArgValue;
-export type MatrixArgValue = (CellValue | undefined)[][];
+export type MatrixArgValue = (CellValue | undefined)[][]; // TODO: replace undefined by null
 export type PrimitiveArgValue = string | number | boolean | null;
 
-export type FunctionReturn = { value: FunctionReturnValue; format?: Format };
+export type FunctionReturn = { value: FunctionReturnValue; format?: FunctionReturnFormat };
 interface FormulaReturn extends Omit<FunctionReturn, "value"> {
   value: FunctionReturnValue | null; // Formulas can return a cell value that can be null for empty cells
-  format?: Format;
+  format?: FunctionReturnFormat;
+}
+export type FunctionReturnValue = ScalarValue | MatrixValue;
```

Comment:
Pretty sure for array formula that `MatrixValue` should be able to contain undefined values.

For example the function transpose can definetevly return undefined values if some cells of the original matrix are empty.

Reply:
I think we have to stay consistent here.

When you refer to an empty cell, e.g. A1:"=B1" with empty B1
A1 --> 0

We should have the same behavior for the matrix result.

That's what Excel does anyway.

I know that the current transpose function don't take this behavior into account. 
This will be done in the associated PR

Reply:
I prefer GSheet results for array functions, but I guess it make sense.